src/main/java/alex/mojaki/s3upload/StreamTransferManager.java

```diff
@@ -399,10 +401,8 @@ public void complete() {
         }
     }
 
-    private void checkCompleteFileIntegrity(String s3ObjectETag) {
-        List<PartETag> parts = new ArrayList<PartETag>(partETags);
```

Comment:
@gkolakowski-ias do you remember if there was any particular reason you chose to make a copy of this list to sort? Do you see any problem with sorting the field in place?

Reply:
I think I found it: https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#synchronizedList(java.util.List)

> It is imperative that the user manually synchronize on the returned list when iterating over it:

Reply:
I'm not sure if that's a problem, but I'm thinking now that the sort method would probably inadvertently invoke the mutex for every little operation, which would slow it down a lot. So I'm going to leave this.